docs/research/ROCA.md

```diff
@@ -0,0 +1,246 @@
+{"type": "mod-hex", "fname": "keys.txt", "idx": 0, "aux": null,
+  "n": "0x94b79a35a5d47040df1503670080a7714ae1ee751aeb32071b3db388b3bf80b11f661c4b8819ebd1c716239c9ec5a202b08a2aa3c17ad6cd17075ba49fcd005d8b8fa50c29433db35c1421727472deddd77bced7e6438db4d447008b11cdb018139bfef2e06c4b4a3e672543a7e9333040fd881815e14b1f1338e90180fd0865",
+  "marked": true, "time_years": 0.16104529886799998, "price_aws_c4": 70.5861544938444}
+(...)
+2019-03-25 18:31:17 [11915] INFO ### SUMMARY ####################
+2019-03-25 18:31:17 [11915] INFO Records tested: 8
+2019-03-25 18:31:17 [11915] INFO .. PEM certs: . . . 0
+2019-03-25 18:31:17 [11915] INFO .. DER certs: . . . 0
+2019-03-25 18:31:17 [11915] INFO .. RSA key files: . 0
+2019-03-25 18:31:17 [11915] INFO .. PGP master keys: 0
+2019-03-25 18:31:17 [11915] INFO .. PGP total keys:  0
+2019-03-25 18:31:17 [11915] INFO .. SSH keys:  . . . 0
+2019-03-25 18:31:17 [11915] INFO .. APK keys:  . . . 0
+2019-03-25 18:31:17 [11915] INFO .. JSON keys: . . . 0
+2019-03-25 18:31:17 [11915] INFO .. LDIFF certs: . . 0
+2019-03-25 18:31:17 [11915] INFO .. JKS certs: . . . 0
+2019-03-25 18:31:17 [11915] INFO .. PKCS7: . . . . . 0
+2019-03-25 18:31:17 [11915] INFO Fingerprinted keys found: 4
+2019-03-25 18:31:17 [11915] INFO WARNING: Potential vulnerability
+2019-03-25 18:31:17 [11915] INFO ################################
+```
+
+It shows that ROCA vulnerability **is present** on this TPM module model. TPM
+firmware update will be required.
+
+Note that ROCA is connected only with RSA, it doesn't affect any other security
+functions, as long as they don't use RSALib.
+
+## Updating TPM firmware
+
+Tools for updating Infineon TPM firmware can be easily found, unfortunately,
+most of them are either UEFI or Windows applications. A Linux port of them can
+be found [here](https://github.com/iavael/infineon-firmware-updater). It
+requires openssl-1.0 (both developer files and runtime library), but it can be
+updated to 1.1.0 version with [this patch](openssl_1_1_0.patch).
+
+First, check if `TPMFactoryUpd` was built successfully and TPM is detected
+properly:
+
+```
+$ ./TPMFactoryUpd -info
+  **********************************************************************
+  *    Infineon Technologies AG   TPMFactoryUpd   Ver 01.01.2459.00    *
+  **********************************************************************
+
+       TPM information:
+       ----------------
+       Firmware valid                    :    Yes
+       TPM family                        :    2.0
+       TPM firmware version              :    5.61.2785.0
+       TPM platformAuth                  :    Empty Buffer
+       Remaining updates                 :    64
+```
+
+Remember the current firmware version number, it will be needed later. Also note
+what is the value of `TPM platformAuth` - it must be `Empty Buffer` in order to
+perform an update. To do this, build and flash coreboot with TPM disabled in
```

Comment:
@krystian-hebel You can also point to a release binaries that has no TPM2.0 support enabled either in coreboot or in SeaBIOS. Refer to CHANGELOGs. I. e. `or use BIOS version prior to v4.8.0.x`